code/__DEFINES/maths.dm

```diff
@@ -0,0 +1,189 @@
+// Credits to Nickr5 for the useful procs I've taken from his library resource.
+
+#define NUM_E 2.71828183
+#define NUM_SQRT2 1.41421356
+
+#define PI						3.1415
+#define SPEED_OF_LIGHT			3e8		//not exact but hey!
+#define SPEED_OF_LIGHT_SQ		9e+16
+#define INFINITY				1e31	//closer then enough
+
+#define SHORT_REAL_LIMIT 16777216
+
+//"fancy" math for calculating time in ms from tick_usage percentage and the length of ticks
+//percent_of_tick_used * (ticklag * 100(to convert to ms)) / 100(percent ratio)
+//collapsed to percent_of_tick_used * tick_lag
+#define TICK_DELTA_TO_MS(percent_of_tick_used) ((percent_of_tick_used) * world.tick_lag)
+#define TICK_USAGE_TO_MS(starting_tickusage) (TICK_DELTA_TO_MS(TICK_USAGE_REAL - starting_tickusage))
+
+#define PERCENT(val) (round(val*100, 0.1))
+#define CLAMP01(x) (CLAMP(x, 0, 1))
+
+//time of day but automatically adjusts to the server going into the next day within the same round.
+//for when you need a reliable time number that doesn't depend on byond time.
+#define REALTIMEOFDAY (world.timeofday + (MIDNIGHT_ROLLOVER * MIDNIGHT_ROLLOVER_CHECK))
+#define MIDNIGHT_ROLLOVER_CHECK ( GLOB.rollovercheck_last_timeofday != world.timeofday ? update_midnight_rollover() : GLOB.midnight_rollovers )
+
+#define SIGN(x) (x!=0 ? x / abs(x) : 0)
+
+#define ATAN2(x, y) ( !x && !y ? 0 : (y >= 0 ? arccos(x / sqrt(x*x + y*y)) : -arccos(x / sqrt(x*x + y*y)) ) )
+
+#define CEILING(x, y) (-round(-x / y) * y)
+
+#define FLOOR(x, y) (round(x / y) * y)
+
+#define CLAMP(CLVALUE,CLMIN,CLMAX) ( max( (CLMIN), min((CLVALUE), (CLMAX)) ) )
+
+#define MODULUS(x, y) (x - y * round(x / y))
+
+// Tangent
+#define TAN(x) (sin(x) / cos(x))
+
+// Cotangent
+#define COT(x) (1 / TAN(x))
+
+#define CSC(x) (1 / sin(x))
+
+#define DEFAULT(a, b) (a ? a : b)
+
+// Greatest Common Divisor - Euclid's algorithm
+/proc/Gcd(a, b)
+	return b ? Gcd(b, a % b) : a
+
+// Least Common Multiple
+#define Lcm(a, b) (abs(a) / Gcd(a, b) * abs(b))
+
+#define INVERSE(x) (1/x)
```

Comment:
Does this define really get used anywhere in the code?

Reply:
no but it's fine to keep, we've got quite a few unused math helpers